ppafm/ocl/field.py

```diff
@@ -409,6 +409,53 @@ def subCores(self, xyzs, Zs, Rcore=0.7, valElDict=None):
         grid = TipDensity(array.T, self.lvec, ctx=self.ctx)
         return grid
 
+    def interp_at(self, lvec_new, shape_new, array_out=None, local_size=(32,), queue=None):
+        '''
+        Interpolate tip density onto a new grid. The tip is assumed to be cented on the origin,
+        so the resizing of the grid happens in the middle of the grid and the corners remain
+        fixed (the origin coordinates of the grids are ignored in the transformation).
+
+        Arguments:
+            lvec_new: array-like of shape (4, 3). Unit cell boundaries for new grid.
+            shape_new: array-like of length 3. New grid shape.
+            array_out: pyopencl.Buffer or None. Output array. If None, then is created automatically.
+            local_size: tuple of a single int. Size of local work group on device.
+            queue: pyopencl.CommandQueue. OpenCL queue on which operation is performed.
+                Defaults to oclu.queue.
+
+        Returns:
+            grid_out: :class:`TipDensity`. New tip density grid.
+        '''
+
+        if len(self.shape) == 4 and self.shape[3] > 1:
+            raise NotImplementedError('Interpolation for 4D grids is not implemented.')
+
+        queue = queue or oclu.queue
+
+        size_new = 4 * np.prod(shape_new[:3])
+        if array_out is None:
+            array_out = cl.Buffer(self.ctx, cl.mem_flags.READ_WRITE, size=size_new)
+        else:
+            assert array_out.size == size_new, f'array size does not match ({array_out.size} != {size_new})'
+        array_out = TipDensity(array_out, lvec_new, shape_new, self.ctx)
+
+        global_size = [int(np.ceil(np.prod(shape_new[:3]) / local_size[0]) * local_size[0])]
+        T = np.concatenate([np.linalg.inv(self.step).T.copy(), np.zeros((3, 1))], axis=1, dtype=np.float32)
+        lvec_in_inv = np.concatenate([np.linalg.inv(self.lvec[1:]), np.zeros((3, 1))], axis=1, dtype=np.float32)
+        dlvec_out = np.concatenate([array_out.step, np.zeros((3, 1))], axis=1, dtype=np.float32)
```

Comment:
I would prefer formating like this (Not just to piss-off Sasha :) )
```
        global_size = [int(np.ceil(np.prod(shape_new[:3]) / local_size[0]) * local_size[0])]
        T           = np.concatenate([np.linalg.inv(self.step).T.copy(), np.zeros((3, 1))], axis=1, dtype=np.float32)
        lvec_in_inv = np.concatenate([np.linalg.inv(self.lvec[1:]),      np.zeros((3, 1))], axis=1, dtype=np.float32)
        dlvec_out   = np.concatenate([array_out.step,                    np.zeros((3, 1))], axis=1, dtype=np.float32)
```

Reply:
Well, I don't have a clear preference either way, but I would rather not spend time worrying too much about the formatting for these kind of almost boilerplate type lines.